monkey/infection_monkey/puppet/plugin_source_extractor.py

```diff
@@ -0,0 +1,113 @@
+import io
+from pathlib import Path
+from tarfile import TarFile, TarInfo
+
+from common.agent_plugins import AgentPlugin
+from common.utils.file_utils import create_secure_directory
+
+
+class PluginSourceExtractor:
+    """
+    Extracts a plugin's source code archive to disk"""
+
+    def __init__(self, plugin_directory: Path):
```

Comment:
I suggest `dst_dir` or `destination_directory`. if you see `PluginSourceExtractor(plugin_directory="/tmp/plugins")` it's not clear if the extractor will take plugins or write them there

Reply:
Good suggestion. I changed it to `plugin_destination_directory`